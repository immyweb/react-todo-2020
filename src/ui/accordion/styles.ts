import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
