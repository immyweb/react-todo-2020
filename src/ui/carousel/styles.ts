import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
