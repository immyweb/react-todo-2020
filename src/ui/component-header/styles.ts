import glamorous from 'glamorous';

import { P, H2 } from '../typography/typography';
import { colours } from '../../styles/colours';

import { mediaQueries } from '../../styles/breakpoints';
import { IAlign } from '../../utils/alignment-utils';

interface IGProps {
  inline?: boolean;
  alignment?: IAlign;
  linkColour?: string;
  background?: boolean;
  hasBackground?: boolean;
}

const styles = {
  fontSizeLarge: 28,
  fontSizeSmall: 24,
  inlineHighlight: 16,
  marginBottomLarge: 12,
  h2LineHeight: 1.14,
  pLineHeight: 1.38,
};

export const TitleWrapper = glamorous.div<IGProps>(
  {
    label: 'title-wrapper',
  },
  ({ alignment, background, theme }: any) => ({
    textAlign: alignment.x,
    background: background ? theme.primary : 'none',
    padding: background ? '13px 20px 13px 15px' : 0,
    borderLeftStyle: background ? 'solid' : 'none',
    borderLeftWidth: '5px',
    borderLeftColor: theme.secondary,
  }),
);

export const Title = glamorous(H2)<IGProps>(
  {
    label: 'title',
    fontSize: styles.fontSizeSmall,
    lineHeight: styles.h2LineHeight,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      lineHeight: styles.h2LineHeight,
    },
  },
  ({ inline, alignment, background }: any) => ({
    display: inline ? 'inline-block' : 'block',
    textTransform: background ? 'unset' : inline ? 'unset' : 'uppercase',
    marginBottom: background ? 0 : styles.marginBottomLarge,
    textAlign: alignment.x,
    color: background ? colours.$white : colours.$offBlack,
  }),
);

export const Highlight = glamorous(Title)<IGProps>(
  {
    label: 'highlight',
    marginBottom: 0,
    fontSize: styles.fontSizeSmall,
    [mediaQueries.tabletPortrait]: {
      fontSize: styles.fontSizeLarge,
      marginBottom: 0,
    },
  },
  ({ inline, alignment, fontColour, background }: any) => ({
    fontSize: inline ? styles.fontSizeSmall : styles.inlineHighlight,
    textTransform: 'uppercase',
    textAlign: alignment.x,
    color: fontColour,
    marginRight: inline ? 4 : 0,
    [mediaQueries.tabletPortrait]: {
      fontSize: inline ? styles.fontSizeLarge : styles.inlineHighlight,
    },
  }),
);

export const SubTitle = glamorous(P)<IGProps>(
  {
    label: 'sub-title',
    lineHeight: styles.pLineHeight,
    textTransform: 'lowercase',
    marginBottom: 20,
    [mediaQueries.tabletPortrait]: {
      lineHeight: styles.pLineHeight,
    },
  },
  ({ alignment, linkColour, hasBackground }: any) => ({
    textAlign: alignment.x,
    marginTop: hasBackground ? 20 : 0,
    '& a': {
      color: linkColour,
      textDecoration: 'none',
    },

    '& a:hover': {
      textDecoration: 'underline',
    },
  }),
);
