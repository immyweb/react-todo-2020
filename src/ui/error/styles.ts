import glamorous from 'glamorous';

import { H3 } from '../typography/typography';
import { fonts } from '../../styles/fonts';

interface IGProps {
  backgroundColour?: string;
  marginBottom?: number;
}

export const ErrorWrapper = glamorous.div<IGProps>(
  {
    label: 'error-wrapper',
    position: 'relative',
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: 200,
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour,
  }),
);

export const Heading = glamorous(H3)<IGProps>(
  {
    fontFamily: fonts.$theSunHeavyCondensed,
    textAlign: 'center',
    fontSize: 22,
  },
  ({ marginBottom }) => ({
    marginBottom: marginBottom ? marginBottom : 5,
  }),
);

export const Graphic = glamorous.div({
  label: 'graphic',
  position: 'relative',
  margin: '0 auto 10px auto',
});
