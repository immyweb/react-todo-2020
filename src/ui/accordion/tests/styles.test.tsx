import * as React from 'react';
import { shallow } from 'enzyme';

import { Border, Link, TextHolder, Item, Header, Title, Panel, Copy } from '../styles';

describe('<Border />', () => {
  it('renders correctly with background colour', () => {
    const wrapper = shallow(<Border colour={'#000000'} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Item />', () => {
  it('renders correctly without border', () => {
    const wrapper = shallow(
      <Item
        open={true}
        index={1}
        theme={{ primary: '#000000', secondary: '#ffffff' }}
        border={false}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with border', () => {
    const wrapper = shallow(
      <Item
        open={true}
        index={1}
        theme={{ primary: '#000000', secondary: '#ffffff' }}
        border={true}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when open', () => {
    const wrapper = shallow(
      <Item open={true} index={1} theme={{ primary: '#000000', secondary: '#ffffff' }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when closed', () => {
    const wrapper = shallow(
      <Item open={false} index={1} theme={{ primary: '#000000', secondary: '#ffffff' }} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<TextHolder />', () => {
  it('renders correctly with no badge', () => {
    const wrapper = shallow(<TextHolder hasBadge={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with badge', () => {
    const wrapper = shallow(<TextHolder hasBadge={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Header />', () => {
  it('renders correctly when opened', () => {
    const wrapper = shallow(<Header open={true} colour={'#000000'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when closed', () => {
    const wrapper = shallow(<Header open={false} colour={'#000000'} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Title />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Title index={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Panel />', () => {
  it('renders correctly when opened', () => {
    const wrapper = shallow(<Panel open={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when closed', () => {
    const wrapper = shallow(<Panel open={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Copy />', () => {
  it('renders correctly with background colour', () => {
    const wrapper = shallow(<Copy colour={'#000000'} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Link />', () => {
  it('renders correctly with background colour', () => {
    const wrapper = shallow(<Link colour={'#000000'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
