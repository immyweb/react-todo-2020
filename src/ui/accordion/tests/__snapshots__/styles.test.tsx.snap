// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Border /> renders correctly with background colour 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(div)
    colour="#000000"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-mncoh0",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-mncoh0",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Copy /> renders correctly with background colour 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(div)
    colour="#000000"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-1n5h0n6",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-1n5h0n6",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Header /> renders correctly when closed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(header)
    colour="#000000"
    open={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-6j9rfl",
    },
    "ref": null,
    "rendered": null,
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-6j9rfl",
      },
      "ref": null,
      "rendered": null,
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Header /> renders correctly when opened 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(header)
    colour="#000000"
    open={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-13bmsvc",
    },
    "ref": null,
    "rendered": null,
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-13bmsvc",
      },
      "ref": null,
      "rendered": null,
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Item /> renders correctly when closed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(div)
    index={1}
    open={false}
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-r83jrh",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-r83jrh",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Item /> renders correctly when open 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(div)
    index={1}
    open={true}
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-qngslt",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-qngslt",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Item /> renders correctly with border 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(div)
    border={true}
    index={1}
    open={true}
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-1i7nohm",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-1i7nohm",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Item /> renders correctly without border 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(div)
    border={false}
    index={1}
    open={true}
    theme={
      Object {
        "primary": "#000000",
        "secondary": "#ffffff",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-qngslt",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-qngslt",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Link /> renders correctly with background colour 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(glamorous(a))
    colour="#000000"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-iye8nw",
    },
    "ref": null,
    "rendered": null,
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-iye8nw",
      },
      "ref": null,
      "rendered": null,
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Panel /> renders correctly when closed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(article)
    open={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-7k84m",
    },
    "ref": null,
    "rendered": null,
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-7k84m",
      },
      "ref": null,
      "rendered": null,
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Panel /> renders correctly when opened 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(article)
    open={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-ceycax",
    },
    "ref": null,
    "rendered": null,
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-ceycax",
      },
      "ref": null,
      "rendered": null,
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<TextHolder /> renders correctly with badge 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(div)
    hasBadge={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-2nnyho",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-2nnyho",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<TextHolder /> renders correctly with no badge 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(div)
    hasBadge={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-sis77o",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-sis77o",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Title /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <glamorous(glamorous(h4))
    index={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "css-hu1kvs",
    },
    "ref": null,
    "rendered": null,
    "type": "h4",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "css-hu1kvs",
      },
      "ref": null,
      "rendered": null,
      "type": "h4",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
